palette/src/hsl.rs

```diff
@@ -21,19 +21,19 @@ use crate::{
 };
 
 /// Linear HSL with an alpha component. See the [`Hsla` implementation in
-/// `Alpha`](struct.Alpha.html#Hsla).
+/// `Alpha`](crate::Alpha#Hsla).
```

Comment:
Oh, do anchors work too? That would be amazing!

Reply:
Yes and no :sweat_smile:  
That's using the `span id=` on the `impl Alpha` block for `Hsl`, so anchors work but they don't seem to be so fine-grained to refer to what we want.
https://github.com/Ogeon/palette/blob/7b8272ee6c25c86960ff54e3766afe786078d0a0/palette/src/hsl.rs#L161

In the Alpha doc, the anchor for the Hsl implementation is `palette/struct.Alpha.html#impl-1`.

I tried searching briefly but couldn't find anything about this. It might need to be suggested as an issue/feature request for rustdoc. It'd be great to have the anchor be the type name instead of `impl-1`, which would mean no more markdown/html spans.

Otherwise, it's an amazing feature and makes cross-linking so much easier. When you use a path that doesn't exist, you get something like a lint warning but the doc build still goes through. I used `cargo doc --no-deps --no-default-features --features std` with an optional `--open` to quickly iterate on builds.

Reply:
Alright, so it's basically the same old issue. :smile: I had even forgotten about it. It's good that they work at all! I don't know if it can be improved that much more in rustdoc, considering you can have multiple `impl` for types and type compositions. But I wouldn't complain if it did. :slightly_smiling_face:

Anyway, let's merge.